core/modules/installer/installer.browser.pages.inc

```diff
@@ -0,0 +1,998 @@
+    '#type' => 'ajax',
+    '#commands' => $commands,
+  );
+}
+
+/**
+ * Validate handler for installer_browser_bulk_install_form().
+ */
+function installer_browser_bulk_install_form_validate($form, &$form_state) {
+  module_load_include('inc', 'installer', 'installer.browser');
+  $type = $form_state['values']['type'];
+
+  // Prepare entered names.
+  $project_names = explode("\n", $form_state['values']['bulk']);
+  $project_names = array_map('trim', $project_names);
+  $project_names = array_filter($project_names, 'strlen');
+  if (empty($project_names)) {
+    form_set_error('bulk', t('Sorry, none of the submitted names are valid.'));
+  }
+  else {
+    foreach ($project_names as $project_name) {
+      // Check not already downloaded.
+      $already_enabled = array();
+      if (backdrop_get_filename($type, $project_name)) {
+        $already_enabled[] = $project_name;
+      }
+      if (!empty($already_enabled)) {
+        form_set_error('bulk', t('The following projects are already installed: %names', array('%names' => implode(', ', $already_enabled))));
+      }
+    }
+  }
+
+  $results = installer_browser_fetch_results(array('items_per_page' => 999, 'type' => $form_state['values']['type']));
+  if (!empty($results)) {
+    $listed_projects = array_keys($results['projects']);
+    $not_found = array_diff($project_names, $listed_projects);
+    if ($not_found) {
+      form_set_error('bulk', t('Sorry, the following projects are not recognized: %names', array('%names' => implode(', ', $not_found))));
+    }
+    else {
+      $projects = array();
+      foreach ($project_names as $project_name) {
+        $projects[$project_name] = $results['projects'][$project_name];
+      }
+      $form_state['projects'] = $projects;
+    }
+  }
+  else {
+    form_set_error('bulk', t('Sorry, the project server is offline.'));
+  }
+}
+
+
+/**
+ * Submit handler for installer_browser_bulk_install_form().
+ */
+function installer_browser_bulk_install_form_submit($form, &$form_state) {
+  module_load_include('inc', 'installer', 'installer.browser');
+  foreach ($form_state['projects'] as $project) {
+    installer_browser_install_queue_add($project);
+  }
+  $form_state['redirect'] = 'admin/modules/installer-browser/' . $form_state['values']['type'];
+}
+
+/**
+ * Page callback: Allows for adding to and removing from the install queue.
+ */
+function installer_browser_install_queue_callback($method, $op, $project_name) {
+  module_load_include('inc', 'installer', 'installer.browser');
+  switch ($op) {
+    case 'add':
+      $projects = installer_browser_get_listed_projects();
+
+      if (isset($projects[$project_name])) {
+        $project = $projects[$project_name];
+        installer_browser_install_queue_add($project);
+      }
+      else {
+        backdrop_set_message(t('Error: The project was not found.'), 'error');
+      }
+      break;
+
+    case 'remove':
+      installer_browser_install_queue_remove($project_name);
+      break;
+  }
```

Comment:
Add `default:` and `return MENU_NOT_FOUND;` for $op other than "add" or "remove".